### PR TITLE
Fix allowing of selecting protocol versions after merge of 1.10.x

### PR DESCRIPTION
**Pull request by ziogaschr**
Keep in mind that Eth protocol version 63 has been dropped.
_EDIT(meowsbits) (And 62 was dropped a long time ago.)_

Ref #365

### CODE REVIEW

File: eth/handler.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ var (
 	syncChallengeTimeout = 15 * time.Second // Time allowance for a node to reply to the sync progress challenge
 )
 
+// DefaultProtocolVersions are the supported versions of the `eth` protocol (first
+// is primary).
+var DefaultProtocolVersions = []uint{eth.ETH66, eth.ETH65, eth.ETH64}
```
Comment (ziogaschr) [inline eth/handler.go:56]: This had to be set in the eth package as the protocol has nested packages logic for (eth|snap) protocols.

File: eth/backend.go [hunk 2]
```diff
@@ -543,7 +543,7 @@ func (s *Ethereum) BloomIndexer() *core.ChainIndexer   { return s.bloomIndexer }
 // Protocols returns all the currently configured
 // network protocols to start.
 func (s *Ethereum) Protocols() []p2p.Protocol {
-	protos := eth.MakeProtocols((*ethHandler)(s.handler), s.networkID, s.ethDialCandidates)
+	protos := eth.MakeProtocols((*ethHandler)(s.handler), s.networkID, s.config.ProtocolVersions, s.ethDialCandidates)
 	if s.config.SnapshotCache > 0 {
 		protos = append(protos, snap.MakeProtocols((*snapHandler)(s.handler), s.snapDialCandidates)...)
```
Comment (ziogaschr) [inline eth/backend.go:548]: At the moment we only have a flag for the eth protocols (66|65|64) and not for the snap protocols. I am not sure if we prefer handling the snap protocol version now or leave them out, as it's just one at the moment.

Comment (meowsbits) [reply eth/backend.go:548]: Noted, thanks for noticing this. SGTM to leave as-is (no flag), then think about adding one when there's more than one to choose from.

**Comment by meowsbits** [comment]
So the `EthProtocols` flag is core-geth specific. (IIRC we added it as a kind of just-in-case option for security cases like Ethereum Classic's Phoenix upgrade where a subset of nodes (OpenEthereum, I think) failed to support 65 properly and nodes supporting (only) lower versions were important in "bridging" the network.) 

This patch fixes that flag to be effectual again, and to support the 66,65,64 set. 

Just making sure I understand the motivation and context here.

**Comment by ziogaschr** [comment]
Yes, you recall correctly. This has been added on core-geth only for this exact reason.

**Comment by meowsbits** [comment]
Note that I've created https://github.com/etclabscore/core-geth/pull/375 in response to the failing tests here; it's just an impatient timeout issue.

**Review by meowsbits — APPROVED** [review]

**Pull request merged**